### ISSUE TITLE
Add a pronunciation audio file to each line

### ISSUE THREAD

**Issue by nicolas-raoul** [anki-decks/anki-deck-for-duolingo-chinese#10]
I think forvo allows such usage, must check. Or Wikimedia Commons.

**Comment by leonfox1** [comment]
Good point.  I am not super familiar with importing audio files, so that will take some research when we get here.

**Comment by nicolas-raoul** [comment]
The audio files could be stored in a folder and committed to Git, for instance `/audio/你.opus`.
Then each Anki note would link to the audio file (might require some coding or using of an addon maybe).

**Comment by leonfox1** [comment]
Well, after creating a test deck with Duolingo sections 1 & 2, I took a stab at adding pronunciation files using a popular Anki add-on tool called [AwesomeTTS](https://ankiatts.appspot.com/).  

The tool is pretty good and offers access to several TTS services, however, of the ones that supported Chinese, several didn't immediately work including Wikitionary.  Wiktionary produced an error "ASCII Codec can't encode character in position 100-101: ordinal not in range(128)".

Google Translate and IMTranslator both produced usable results, but Google Translate is "heavily rate-limited and not recommended for mass generation".  Basically, when I use it for multiple files at the same time, it has to "sleep for 30 seconds" in between each one.  

IMTranslator, on the other hand, appears to be working quite well, though the auto-downloader appears to sleep for 30 seconds every 5 uses.  Also, it is unclear whether we are allowed to use it for our purposes--license says "personal, non-commerical use".  I did a test download of the first 33 vocab words, all of which are good quality.

Beyond that, I didn't see any successful have TTS services to auto-download with this Anki add-on.  Thoughts?

**Comment by nicolas-raoul** [comment]
We may have to settle for "non-commercial" for now. We are clearly not "personal", though :-/

In the future it would be great if all content was [open](https://en.wikipedia.org/wiki/Open_content), we can create a separate issue about that.

Forvo pronunciations [come under a CC BY-NC-SA 3.0 license](https://forvo.com/terms-and-conditions/), which means that downloading them from the website and putting them on GitHub under the same license is totally lawful. Downloading each pronunciation is rather time-consuming, so paying [2 USD](https://api.forvo.com/plans-and-pricing/) to use the API once could be an idea.

**Comment by leonfox1** [comment]
Gotcha. I will take a look at Forva tomorrow. Thanks!

**Comment by leonfox1** [comment]
Just went to check out for the $2 per month Forvo plan and was slightly annoyed to find out that the only options are a 6 month of 1 year subscription.  So it will probably cost an extra $10 to download the audio. 
 Anyways, just thought I would mention it.

**Comment by nicolas-raoul** [comment]
Oh :-(

The tragic thing is, most people who upload pronunciations to Forvo think it becomes open data...
https://opendata.stackexchange.com/questions/4329/no-data-dump-and-low-daily-cap-api-can-it-be-called-open-data

**Comment by nicolas-raoul** [comment]
This project might be a better option: http://shtooka.net/overview.php?lang=cmn
Easily downloadable dumps containing all audio files are available: http://shtooka.net/download.php

**Comment by leonfox1** [comment]
Nice. I saw that when I was doing some searches, but didn't look at it closely and didn't check the Chinese on it.  I will check it out tomorrow. I am going to try and upload an updated APKG file tonight.  I was almost there, but then got distracted by something.  Anyways, thanks for this find.
